plugins/s3-viewer-backend/src/types.ts

```diff
@@ -8,7 +8,7 @@ export type S3Platform = {
   endpoint: string;
   endpointName: string;
   region: string;
-  credentials: {
+  credentials?: {
```

Comment:
Does this being optional now need new code elsewhere to ensure that the credentials are available for the other providers?

Reply:
I thought about it, but not really. The `S3` client has an optional `credentials`, so it can be optional. The only place where it might be failing is in the rados-gw type in the `SignatureV4` method. However, we are sure it won't be `undefined` there because in the constructor the config requires the credentials to be set and not empty.